File: android/src/main/kotlin/com/fit321/fitui/components/FitAvailabilityDay.kt
package com.fit321.fitui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Close
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.fit321.fitui.theme.LocalFitTheme
import com.fit321.fitui.tokens.FitColors
import com.fit321.fitui.tokens.FitFont
import com.fit321.fitui.tokens.FitRadius
import com.fit321.fitui.tokens.FitSize
import com.fit321.fitui.tokens.FitSpacing

// ============================================================================
// FitAvailabilityDay — coach availability row for one day-of-week.
// Mirrors Swift FitAvailabilityDay. See `docs/components.md`
// § FitAvailabilityDay.
// ============================================================================

enum class FitDayOfWeek(val label: String) {
    Monday("Monday"),
    Tuesday("Tuesday"),
    Wednesday("Wednesday"),
    Thursday("Thursday"),
    Friday("Friday"),
    Saturday("Saturday"),
    Sunday("Sunday")
}

data class FitAvailabilityInterval(val start: String, val end: String)

@Composable
fun FitAvailabilityDay(
    day: FitDayOfWeek,
    isActive: Boolean,
    onActiveChange: (Boolean) -> Unit,
    intervals: List<FitAvailabilityInterval>,
    onAddInterval: (() -> Unit)? = null,
    onRemoveInterval: ((Int) -> Unit)? = null,
    onIntervalEdit: ((Int) -> Unit)? = null,
    validationError: String? = null,
    modifier: Modifier = Modifier
) {
    val theme = LocalFitTheme.current
    val leftIndent = FitSize.checkboxSize + FitSpacing.sp3

    Column(
        modifier = modifier.fillMaxWidth(),
        verticalArrangement = Arrangement.spacedBy(FitSpacing.sp3)
    ) {
        // Header
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(FitSpacing.sp3)
        ) {
            FitCheckbox(
                checked = isActive,
                onCheckedChange = onActiveChange
            )
            Text(
                day.label,
                style = FitFont.body1.copy(fontWeight = FontWeight.Medium),
                color = if (isActive) theme.textPrimary else theme.textTertiary
            )
        }

        if (isActive) {
            Column(
                modifier = Modifier.padding(start = leftIndent),
                verticalArrangement = Arrangement.spacedBy(FitSpacing.sp2)
            ) {
                intervals.forEachIndexed { idx, interval ->
                    IntervalRow(
                        interval = interval,
                        index = idx,
                        onEdit = onIntervalEdit,
                        onRemove = onRemoveInterval
                    )
                }
                if (onAddInterval != null) {
                    Row(
                        modifier = Modifier.clickable(onClick = onAddInterval),
                        verticalAlignment = Alignment.CenterVertically,
                        horizontalArrangement = Arrangement.spacedBy(FitSpacing.sp2)
                    ) {
                        Icon(
                            Icons.Default.Add, null,
                            tint = FitColors.brandPrimary,
                            modifier = Modifier.size(14.dp)
                        )
                        Text("Add interval", style = FitFont.body2, color = FitColors.brandPrimary)
                    }
                }
            }
        }

        if (validationError != null) {
            Text(
                validationError,
                style = FitFont.footnote,
                color = FitColors.Red.r400,
                modifier = Modifier.padding(start = leftIndent)
            )
        }
    }
}

@Composable
private fun IntervalRow(
    interval: FitAvailabilityInterval,
    index: Int,
    onEdit: ((Int) -> Unit)?,
    onRemove: ((Int) -> Unit)?
) {
    val theme = LocalFitTheme.current
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(FitSpacing.sp2)
    ) {
        TimeInputBox(value = interval.start, onClick = { onEdit?.invoke(index) })
        Text("to", style = FitFont.body2, color = theme.textSecondary)
        TimeInputBox(value = interval.end, onClick = { onEdit?.invoke(index) })

        Spacer(Modifier.weight(1f))

        if (onRemove != null) {
            Box(
                modifier = Modifier
                    .size(28.dp)
                    .clip(CircleShape)
                    .background(FitColors.error.copy(alpha = 0.10f))
                    .clickable { onRemove(index) },
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    Icons.Default.Close, null,
                    tint = FitColors.error,
                    modifier = Modifier.size(12.dp)
                )
            }
        }
    }
}

@Composable
private fun TimeInputBox(value: String, onClick: () -> Unit) {
    val theme = LocalFitTheme.current
    Box(
        modifier = Modifier
            .size(56.dp)
            .clip(RoundedCornerShape(FitRadius.input))
            .background(theme.surfaceLow)
            .clickable(onClick = onClick),
        contentAlignment = Alignment.Center
    ) {
        Text(
            value,
            style = FitFont.body2.copy(fontWeight = FontWeight.Medium),
            color = theme.textPrimary
        )
    }
}

File: Sources/FitUI/Components/FitAvailabilityDay.swift
import SwiftUI

// MARK: - FitAvailabilityDay
//
// Coach availability row for one day-of-week — header (FitCheckbox +
// day name) + collapsible list of time intervals + optional add /
// remove buttons + optional validation error text. Used in coach
// onboarding (availability step) and Settings → Availability.
// See `docs/components.md` § FitAvailabilityDay.

public enum FitDayOfWeek: Int, CaseIterable, Identifiable, Hashable {
    case monday = 1, tuesday, wednesday, thursday, friday, saturday, sunday

    public var id: Int { rawValue }

    public var label: String {
        switch self {
        case .monday: return "Monday"
        case .tuesday: return "Tuesday"
        case .wednesday: return "Wednesday"
        case .thursday: return "Thursday"
        case .friday: return "Friday"
        case .saturday: return "Saturday"
        case .sunday: return "Sunday"
        }
    }
}

/// Free-form interval so callers can use whatever time type fits
/// (`Date`, `(Int,Int)`, etc.); the component just renders strings
/// supplied by the caller. Keeps the component decoupled from
/// platform-specific Date types.
public struct FitAvailabilityInterval: Hashable {
    public var start: String
    public var end: String

    public init(start: String, end: String) {
        self.start = start
        self.end = end
    }
}

public struct FitAvailabilityDay: View {
    let day: FitDayOfWeek
    @Binding var isActive: Bool
    @Binding var intervals: [FitAvailabilityInterval]
    let onAddInterval: (() -> Void)?
    let onRemoveInterval: ((Int) -> Void)?
    let onIntervalEdit: ((Int) -> Void)?
    let validationError: String?

    @Environment(\.fitTheme) private var theme

    public init(
        day: FitDayOfWeek,
        isActive: Binding<Bool>,
        intervals: Binding<[FitAvailabilityInterval]>,
        onAddInterval: (() -> Void)? = nil,
        onRemoveInterval: ((Int) -> Void)? = nil,
        onIntervalEdit: ((Int) -> Void)? = nil,
        validationError: String? = nil
    ) {
        self.day = day
        self._isActive = isActive
        self._intervals = intervals
        self.onAddInterval = onAddInterval
        self.onRemoveInterval = onRemoveInterval
        self.onIntervalEdit = onIntervalEdit
        self.validationError = validationError
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: FitSpacing.sp3) {
            // Header
            HStack(spacing: FitSpacing.sp3) {
                FitCheckbox(checked: $isActive)
                Text(day.label)
                    .font(.custom(FitFont.family, size: 16).weight(.medium))
                    .foregroundColor(isActive ? theme.textPrimary : theme.textTertiary)
                Spacer()
            }

            // Intervals (visible only when active)
            if isActive {
                VStack(alignment: .leading, spacing: FitSpacing.sp2) {
                    ForEach(intervals.indices, id: \.self) { idx in
                        intervalRow(idx)
                    }

                    if let onAddInterval {
                        Button(action: onAddInterval) {
                            HStack(spacing: FitSpacing.sp2) {
                                Image(systemName: "plus")
                                    .font(.system(size: 14, weight: .medium))
                                Text("Add interval")
                                    .font(FitFont.body2)
                            }
                            .foregroundColor(FitColors.brandPrimary)
                        }
                        .buttonStyle(.plain)
                    }
                }
                .padding(.leading, FitSize.checkboxSize + FitSpacing.sp3)
            }

            if let validationError {
                Text(validationError)
                    .font(.custom(FitFont.family, size: 13))
                    .foregroundColor(FitColors.Red.r400)
                    .padding(.leading, FitSize.checkboxSize + FitSpacing.sp3)
            }
        }
    }

    @ViewBuilder
    private func intervalRow(_ idx: Int) -> some View {
        HStack(spacing: FitSpacing.sp2) {
            timeInputBox(intervals[idx].start, idx: idx)

            Text("to")
                .font(FitFont.body2)
                .foregroundColor(theme.textSecondary)

            timeInputBox(intervals[idx].end, idx: idx)

            Spacer(minLength: 0)

            if let onRemoveInterval {
                Button(action: { onRemoveInterval(idx) }) {
                    Image(systemName: "xmark")
                        .font(.system(size: 12, weight: .semibold))
                        .foregroundColor(FitColors.error)
                        .frame(width: 28, height: 28)
                        .background(FitColors.error.opacity(0.1))
                        .clipShape(Circle())
                }
                .buttonStyle(.plain)
            }
        }
    }

    @ViewBuilder
    private func timeInputBox(_ value: String, idx: Int) -> some View {
        Button(action: { onIntervalEdit?(idx) }) {
            Text(value)
                .font(.custom(FitFont.family, size: 14).weight(.medium))
                .foregroundColor(theme.textPrimary)
                .frame(width: 56, height: 56)
                .background(theme.surfaceLow)
                .clipShape(RoundedRectangle(cornerRadius: FitRadius.input))
        }
        .buttonStyle(.plain)
    }
}
